src/python/tests/gff3/test_extract_annotation.py

```diff
@@ -22,15 +22,230 @@
 
 """
 
-from typing import Optional
+from contextlib import nullcontext as does_not_raise
+from typing import ContextManager, Optional
 import pytest
+from pytest import raises
 
-from ensembl.io.genomio.gff3.extract_annotation import FunctionalAnnotations
+from Bio.SeqFeature import SeqFeature
+
+from ensembl.io.genomio.gff3.extract_annotation import (
+    FunctionalAnnotations,
+    MissingParentError,
+    AnnotationError,
+)
 
 
 class TestFunctionalAnnotations:
     """Tests for the `FunctionalAnnotations` class."""
 
+    def test_init(self):
+        """Tests the `FunctionaAnnotation.__init__()` method."""
+        annot = FunctionalAnnotations()
+        assert annot
+
+    @pytest.mark.parametrize(
+        "seq_feat_type, feat_type, expected",
+        [
+            ("gene", "gene", does_not_raise()),
+            ("mRNA", "transcript", does_not_raise()),
+            ("CDS", "translation", does_not_raise()),
+            ("transposable_element", "transposable_element", does_not_raise()),
+            ("gene", "bad_type", raises(KeyError)),
+        ],
+    )
+    def test_add_feature_no_parent(self, seq_feat_type: str, feat_type: str, expected: ContextManager):
+        """Tests the `FunctionaAnnotation.add_feature()` method with only one feature.
+
+        Args:
+            seq_feat_type: Type for the SeqFeature to add.
+            feat_type: Type category for that SeqFeature.
+
+        """
+        annot = FunctionalAnnotations()
+        with expected:
+            feature = SeqFeature(type=seq_feat_type, id="featA")
```

Comment:
Avoid having code that does not belong to your codebase inside the context manager to avoid an unexpected exception making your test pass, e.g. if this line was to raise a `KeyError` in your last parametrization the test would pass successfully without even running the `add_feature()` part.